services/proxy/README.md

```diff
@@ -50,6 +50,78 @@ unprotected: false # with false (default), calling the endpoint requires authori
+  username never changes.
+
+### Configuration
+
+To enable automatic user provisioning, the following environment variables must
+be set for the proxy service:
+
+* `PROXY_AUTOPROVISION_ACCOUNTS`: Set to `true` to enable automatic user provisioning.
+* `PROXY_AUTOPROVISION_CLAIM_USERNAME`: The name of an OIDC claim whose value
+  should be used as the username for the autoprovsioned user in ownCloud
+  Infinite Scale. Defaults to `preferred_username`. Can also be set to e.g.
+  `sub` to guarantee a unique and stable username.
+* `PROXY_AUTOPROVISION_CLAIM_EMAIL`: The name of an OIDC claim whose value
+  should be used for the `mail` attribute of the autoprovisioned user in
+  ownCloud Infinite Scale. Defaults to `email`.
+* `PROXY_AUTOPROVISION_CLAIM_DISPLAYNAME`: The name of an OIDC claim whose
+  value should be used for the `displayname` attribute of the autoprovisioned
+  user in ownCloud Infinite Scale. Defaults to `name`.
+* `PROXY_AUTOPROVISION_CLAIM_GROUPS`: The name of an OIDC claim whose value
+  should be used to maintain a user's group membership. The claim value should
+  contain a list of group names the user should be a member of. Defaults to
+  `groups`.
+* `PROXY_USER_OIDC_CLAIM`: When resolving and authenticated OIDC user, the
+  value of this claims is used to lookup the user in the users service. For
+  auto provisioning setups this usually is the same claims as set via
+  `PROXY_AUTOPROVISION_CLAIM_USERNAME`.
+* `PROXY_USER_CS3_CLAIM`: This is the name of the user attribute in ocis that
+  is used to lookup the user by the value of the `PROXY_USER_OIDC_CLAIM`. For
+  auto provisioning setups this usually needs to be set to `username`.
+
+### How it works
+
+When a user logs into ownCloud Infinite Scale for the first time, the proxy
+checks if that user already exists by querying the `users` service for users
+where the attribute set in `PROXY_USER_CS3_CLAIM` matches the value of the OIDC
+claim configure in `PROXY_USER_OIDC_CLAIM`.
+
+If the users does not exist, the proxy will create a new user via the `graph`
+service using the claim values configured in
+`PROXY_AUTOPROVISION_CLAIM_USERNAME`, `PROXY_AUTOPROVISION_CLAIM_EMAIL` and
+`PROXY_AUTOPROVISION_CLAIM_DISPLAYNAME`.
+
+If the user does already exist, the proxy will check if the user's email or
+displayname has changed and update those accordingly via `graph` service.
+
+Next, the proxy will check if the user is a member of the groups configured in
+`PROXY_AUTOPROVISION_CLAIM_GROUPS`. It will add the user to the groups listed
+in there and remove it from all other groups that it is currently a member of.
+Groups that do not exist yet will be created. Note: This can be a somewhat
+costly operation, especially if the user is a member of a large number of
+groups. If the group memberships of a user are changed in the IDP after the
+first login it can take up to 5 minutes until the changes are reflected in
+ownCloud Infinite Scale.
```

Comment:
```suggestion
Next, the proxy will check if the user is a member of the groups configured in
`PROXY_AUTOPROVISION_CLAIM_GROUPS`. It will add the user to the groups listed
in the envvar and removes it from all other groups that he is currently a member of.
Groups that do not exist in the external IDP yet will be created. Note: This can be a
somewhat costly operation, especially if the user is a member of a large number of
groups. If the group memberships of a user are changed in the IDP after the
first login, it can take up to 5 minutes until the changes are reflected in Infinite Scale.
```